cppcache/src/CacheFactory.cpp

```diff
@@ -87,76 +73,83 @@ CacheFactory::CacheFactory(const std::shared_ptr<Properties> dsProps) {
   pdxReadSerialized = false;
   this->dsProp = dsProps;
 }
- std::shared_ptr<Cache> CacheFactory::create() {
-   ACE_Guard<ACE_Recursive_Thread_Mutex> connectGuard(*g_disconnectLock);
 
-   LOGFINE("CacheFactory called DistributedSystem::connect");
-   auto cache = create(DEFAULT_CACHE_NAME, nullptr);
+Cache CacheFactory::create() {
+  ACE_Guard<ACE_Recursive_Thread_Mutex> connectGuard(*g_disconnectLock);
 
-   auto& cacheImpl = cache->m_cacheImpl;
-   const auto& serializationRegistry = cacheImpl->getSerializationRegistry();
-   const auto& pdxTypeRegistry = cacheImpl->getPdxTypeRegistry();
-   const auto& memberListForVersionStamp =
-       std::ref(*(cacheImpl->getMemberListForVersionStamp()));
+  LOGFINE("CacheFactory called DistributedSystem::connect");
+  auto cache = create(DEFAULT_CACHE_NAME, nullptr);
 
-   serializationRegistry->addType2(
-       std::bind(TXCommitMessage::create, memberListForVersionStamp));
+  auto& cacheImpl = cache.m_cacheImpl;
+  const auto& serializationRegistry = cacheImpl->getSerializationRegistry();
```

Comment:
I have been using `auto&&` over explicit `const auto&` a lot lately and I like it.Thoughts?

```auto&& serializationRegistry = cacheImpl->getSerializationRegistry();```
If the method returns a const then the variable is const, otherwise it isn't. Maybe it ins't the best choice but would love to hear arguments.

Reply:
Are you sure it propagates const-ness?  Below compile under clang c++11/14.

```cpp
const int f() { return 42; }
int main() {
    auto&& i = f();
    return ++i;
}
```

Reply:
In this case no, but in this it does.
```
class foo {
    public:
    std::string str;
    const std::string& getString() {return str;}
}

int main() {
    auto&& str = foo{}.getString();
    return str.append("bar");
}
``` 
In your case the `const int` is value return. I don't think there is any value in the const being applied to the lvalue since its a copy anyway.

Reply:
That example helped. Looks like value of `const auto&` is keeping local scoped variable const.  I haven't really formulated an opinion on this yet and right now I can go either way.